accelerator/BUILD

```diff
@@ -5,25 +5,26 @@ load("@io_bazel_rules_docker//container:container.bzl", "container_push")
 
 cc_binary(
     name = "accelerator",
-    srcs = [
-        "config.c",
-        "config.h",
-    ] + select({
+    srcs = select({
         "//connectivity/mqtt:mqtt_enable": [
             "conn_mqtt.c",
         ],
         "//conditions:default": ["main.c"],
     }),
     copts = [
         "-DLOGGER_ENABLE",
-        "-DMQTT_ENABLE",
     ] + select({
         ":DEBUG_MODE": ["-g"],
         ":PROFILING_MODE": [
             "-DNDEBUG",
             "-pg",
         ],
         "//conditions:default": ["-DNDEBUG"],
+    }) + select({
+        "//connectivity/mqtt:mqtt_enable": [
+            "-DMQTT_ENABLE",
```

Comment:
Should we add `MQTT_ENABLE` in this rule? Since `main.c` does not exist the `MQTT_ENABLE` conditional group.

Reply:
The depending files of `main.c` and `conn_mqtt.c` (i.e. cocfig.h) would change depend on the `MQTT_ENABLE` building identifier.

Reply:
Is removing `-DMQTT_ENABLE` in this rule causing further errors? I think it's wierd to place the copts of the dependencies here (which is already added in `ta_config` rule)

Reply:
Is the problem that ta_core_t in different translation units(`conn_mqtt.c and config.c`) has different layout? Since `conn_mqtt.c` does not define MQTT_ENQBLE but `config.c` does.

Reply:
I tried to build without MQTT_ENABLE here last night
main.c worked fine, but when conn_mqtt.c includes client_common.h it failed. client_common.h has been guarded by MQTT_ENABLE so conn_mqtt.c would see an empty file there